File: examples/javascript/basic_gathering_loop.js
const server = 'https://api.artifactsmmo.com';
const token = "TOKEN"; //Put your token here
const character = "YOUR_CHARACTER_NAME"; //Put your character name here
let cooldown;
let timeout;

//This script is an example of how to loop each time cooldown is complete.
function performGathering() {
  const url = server + '/my/' + character + '/action/gathering';

  const headers = {
    'Content-Type': 'application/json',
    Accept: 'application/json',
    Authorization: 'Bearer ' + token,
  };

  return fetch(url, {
    method: 'POST',
    headers: headers,
  }).then((gatheringResponse) => {
    if (gatheringResponse.status === 498) {
      console.log('The character cannot be found on your account.');
      return;
    }
    else if (gatheringResponse.status === 497) {
      console.log("Your character's inventory is full.");
      return;
    }
    else if  (gatheringResponse.status === 499) {
      console.log('Your character is in cooldown.');
      return;
    }
    else if  (gatheringResponse.status === 493) {
      console.log('The resource is too high-level for your character.');
      return;
    }
    else if (gatheringResponse.status === 598) {
      console.log('No resource on this map.');
      return;
    }
    else if (gatheringResponse.status !== 200) {
      console.log('An error occurred while gathering the resource.');
      return;
    }

    if (gatheringResponse.status === 200) {
        gatheringResponse.json().then((data) => {
            console.log('Your character successfully gathered the resource.');
            cooldown = data.data.cooldown.totalSeconds;
            setTimeout(performGathering, cooldown * 1000);
      });

    }
      });
}


performGathering()

File: examples/python/basic_gathering_loop.py
import requests
import time

#Server url
server = "https://api.artifactsmmo.com"
#Your account token (https://artifactsmmo.com/account)
token = "YOUR_TOKEN_HERE"
#Name of yout character
character = "CHARACTER_NAME" 
cooldown = 5

def perform_gathering() -> int:
    url = f"{server}/my/{character}/action/gathering"

    headers = {
        "Content-Type": "application/json",
        "Accept": "application/json",
        "Authorization": f"Bearer {token}"
    }
    gathering_response  = requests.post(url, headers=headers)

    if gathering_response.status_code == 498:
        print("The character cannot be found on your account.")
        exit()
    elif gathering_response.status_code == 497:
        print("Your character's inventory is full.")
        exit()
    elif gathering_response.status_code == 499:
        print("Your character is in cooldown.")
    elif gathering_response.status_code == 493:
        print("The resource is too high-level for your character.")
        exit()
    elif gathering_response.status_code != 200:
        print("An error occured while gathering the ressource.")
        exit()
    else:
        data = gathering_response.json()["data"]
        print("Your character successfully gathered the ressource.")

        #Return the cooldown in seconds
        return data["cooldown"]["totalSeconds"]

#Loop
while True:
    cooldown = perform_gathering()

    # Wait for the cooldown
    time.sleep(cooldown)